### PR TITLE
Feat: added support for custom maven arguments

### PR DESCRIPTION
**Pull request by ShiriV**
- [x] Tests written and linted [ℹ︎](https://github.com/snyk/general/wiki/Tests)
- [ ] Documentation written [ℹ︎](https://github.com/snyk/general/wiki/Documentation)
- [X] Commit history is tidy [ℹ︎](https://github.com/snyk/general/wiki/Git)

### What this does

Adding the capability to pass args after the double dash to the maven commands being run inside the java call graph builder.

### More information

- [Jira ticket FLOW-692](https://snyksec.atlassian.net/jira/software/projects/FLOW/boards/96?selectedIssue=FLOW-692)

### CODE REVIEW

File: lib/mvn-wrapper-legacy.ts [hunk 1]
```diff
@@ -65,20 +65,26 @@ export function mergeMvnClassPaths(classPaths: string[]): string {
     .join(path.delimiter);
 }
 
-export async function getClassPathFromMvn(targetPath: string): Promise<string> {
+export async function getClassPathFromMvn(
+  targetPath: string,
+  customMavenArgs?: string[],
```
Comment (darmalovan) [inline lib/mvn-wrapper-legacy.ts:70]: You can have custom args with a default value of `[]`, and then remove line 74 ?

File: lib/mvn-wrapper.ts [hunk 2]
```diff
@@ -44,29 +44,51 @@ function runCommand(projectDirectory: string, args: string[]): Promise<string> {
 async function evaluateExpression(
   projectDirectory: string,
   expression: string,
+  customMavenArgs?: string[],
```
Comment (darmalovan) [inline lib/mvn-wrapper.ts:47]: Same here? 
https://github.com/snyk/java-call-graph-builder/pull/51/files#r626540101

File: lib/mvn-wrapper-legacy.ts (hunk 1 above)
Comment (ShiriV) [reply lib/mvn-wrapper-legacy.ts]: Done

File: lib/mvn-wrapper.ts (hunk 2 above)
Comment (ShiriV) [reply lib/mvn-wrapper.ts]: Done

**Review by darmalovan — APPROVED** [review]

**Pull request merged**